pkg/splunk/enterprise/configuration.go

```diff
@@ -829,51 +863,54 @@ func updateSplunkPodTemplateWithConfig(ctx context.Context, client splcommon.Con
+	scopedLog.Info("LivenessProbe", "Configured", livenessProbe)
+	return livenessProbe
 }
 
-// getReadinessProbe provides the probe for checking the readiness of the Pod
-// pod is ready if container artifact file is created with contents of "started".
-func getReadinessProbe(ctx context.Context, cr splcommon.MetaObject, instanceType InstanceType, spec *enterpriseApi.CommonSplunkSpec, additionalDelay int32) *corev1.Probe {
+// getReadinessProbe the probe for checking the readiness of the Pod
+// uses script provided by enterprise container to check if pod is ready
+func getReadinessProbe(ctx context.Context, cr splcommon.MetaObject, instanceType InstanceType, spec *enterpriseApi.CommonSplunkSpec) *corev1.Probe {
 	reqLogger := log.FromContext(ctx)
-	scopedLog := reqLogger.WithName("getReadinessProbe")
-
-	readinessDelay := int32(readinessProbeDefaultDelaySec)
-
-	// If configured, always use the readiness initial delay from the CR
-	if spec.ReadinessInitialDelaySeconds != 0 {
-		readinessDelay = spec.ReadinessInitialDelaySeconds
-	} else {
-		readinessDelay += additionalDelay
-	}
-
-	scopedLog.Info("ReadinessProbeInitialDelay", "configured", spec.ReadinessInitialDelaySeconds, "additionalDelay", additionalDelay, "finalCalculatedValue", readinessDelay)
+	scopedLog := reqLogger.WithName("getReadinessProbe").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
+	readinessProbe := getProbeWithConfigUpdates(&defaultReadinessProbe, spec.ReadinessProbe, spec.ReadinessInitialDelaySeconds)
+	scopedLog.Info("ReadinessProbe", "Configured", readinessProbe)
+	return readinessProbe
+}
 
-	readinessCommand := []string{
-		"/bin/grep",
-		"started",
-		"/opt/container_artifact/splunk-container.state",
+func getProbeWithConfigUpdates(defaultProbe *corev1.Probe, configuredProbe *corev1.Probe, configuredDelay int32) *corev1.Probe {
+	if configuredProbe != nil {
+		// Always take a separate probe, instead of referring the memory address from spec.
+		// (Referring the configured Probe memory is kind of OK as we are not writing to the DB, however
+		// updating any values(if the Application needs to do) can cause confusion when referring the CR
+		// while handling a reconcile event)
+		var derivedProbe = *configuredProbe
+		if derivedProbe.InitialDelaySeconds == 0 {
+			if configuredDelay != 0 {
+				derivedProbe.InitialDelaySeconds = configuredDelay
+			} else {
+				derivedProbe.InitialDelaySeconds = defaultProbe.InitialDelaySeconds
+			}
+		}
+		if derivedProbe.TimeoutSeconds == 0 {
+			derivedProbe.TimeoutSeconds = defaultProbe.TimeoutSeconds
+		}
+		if derivedProbe.PeriodSeconds == 0 {
+			derivedProbe.PeriodSeconds = defaultProbe.PeriodSeconds
+		}
+		if derivedProbe.Exec == nil {
+			derivedProbe.Exec = defaultProbe.Exec
+		}
+		return &derivedProbe
+	} else if configuredProbe == nil && configuredDelay != 0 {
+		var derivedProbe = *defaultProbe
+		derivedProbe.InitialDelaySeconds = configuredDelay
```

Comment:
Should we have the `derviedProbe.InitialDelaySeconds = configuredDelay` regardless of configuredProbe to duplicate the assignment ? Seems like that's what we are trying to achieve here?

Reply:
That would ignore the scenario when configuredDelay = 0 however deafult Probe has non zero delay.